### PR TITLE
DO NOT MERGE: migration from AWS storage to Azure storage

### PR DESCRIPTION
**Pull request by cjoshi-who**
**DO NOT MERGE THIS**
This is just to compare the changes made across versions and it is not to be merged

**Description**
This PR removes usage of AWS S3 storage and adds usage of Azure storage.

- Renamed ecs_deployment to azure_deployment
- Removed aws folder and usages
- Renamed aws or ecs file to Azure files
- Added settings for Azure storage settings 

**Following are the variables added**

- AZURE_ACCOUNT_NAME
- AZURE_ACCOUNT_KEY
- AZURE_CONNECTION_STRING
- AZURE_CONTAINER

**ScreenShots**

image

image

image

image

### CODE REVIEW

File: .docker/azure_deployment/Dockerfile.prod.azure [hunk 1]
```diff
@@ -24,19 +24,17 @@ COPY requirements.txt                   /badgr_server
...
+COPY .docker/azure_deployment/uwsgi.ini              /badgr_server
+COPY .docker/azure_deployment/wsgi.py                /badgr_server
 COPY apps                               /badgr_server/apps
-COPY .docker/ecs_deployment/daemon.sh /badgr_server
+COPY .docker/azure_deployment/daemon.sh /badgr_server
 
-COPY .docker/ecs_deployment/aws /badgr_server/apps/mainsite/aws
-
-COPY .docker/ecs_deployment/settings_local.prod.ecs.py /badgr_server/apps/mainsite/settings_local.py
+COPY .docker/azure_deployment/settings_local.prod.ecs.py /badgr_server/apps/mainsite/settings_local.py
```
Comment (krishnamadhavan) [inline .docker/azure_deployment/Dockerfile.prod.azure:37]: Can we rename this file and its usages to `settings_local.prod.py` removing the term `ecs` ?

File: .docker/azure_deployment/settings_local.prod.ecs.py [hunk 2]
```diff
@@ -141,4 +140,11 @@
...
-OPEN_FOR_SIGNUP = False
\ No newline at end of file
+OPEN_FOR_SIGNUP = False
+
+# Azure storage settings
+
+AZURE_ACCOUNT_NAME = os.getenv('AZURE_ACCOUNT_NAME')
+AZURE_ACCOUNT_KEY = os.getenv('AZURE_ACCOUNT_KEY')
+AZURE_CONNECTION_STRING = os.getenv('AZURE_CONNECTION_STRING')
+AZURE_CONTAINER = os.getenv('AZURE_CONTAINER')
```
Comment (krishnamadhavan) [inline .docker/azure_deployment/settings_local.prod.ecs.py:150]: Can you add a new line at the end of file ? It's recommended practise.

File: apps/mainsite/settings.py [hunk 3]
```diff
@@ -497,3 +497,10 @@
 SAML_EMAIL_KEYS = ['Email', 'email', 'mail', 'emailaddress', 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress']
 SAML_FIRST_NAME_KEYS = ['FirstName', 'givenName', 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname']
 SAML_LAST_NAME_KEYS = ['LastName', 'sn', 'surname', 'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname']
+
+# AZURE SETTINGS
+DEFAULT_FILE_STORAGE = 'storages.backends.azure_storage.AzureStorage'
+AZURE_ACCOUNT_NAME = 'YOUR_ACCOUNT_NAME_HERE'
+AZURE_ACCOUNT_KEY = 'YOUR_ACCOUNT_KEY_HERE'
+AZURE_CONNECTION_STRING = 'YOUR_CONNECTION_STRING_HERE'
+AZURE_CONTAINER = 'CONTAINER_NAME_HERE'
```
Comment (krishnamadhavan) [inline apps/mainsite/settings.py:506]: Is this used only in local or in production ?

File: apps/mainsite/settings_local.py.example [hunk 4]
```diff
@@ -104,7 +104,7 @@ EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
 # AZURE_ACCOUNT_KEY = ''
 # MEDIA_URL = 'http://azureaccountname.blob.core.windows.net/'
 # AZURE_CONTAINER = 'mediafiles'
-
+# AZURE_CONNECTION_STRING = ''
```
Comment (krishnamadhavan) [inline apps/mainsite/settings_local.py.example:107]: Can we remove this comment out code ?

File: requirements.txt [hunk 5]
```diff
@@ -5,11 +5,12 @@ pytz==2017.2
 # PIL support
 Pillow==7.1.0
 python-dateutil==2.7.5
-
 # mysql database
 mysqlclient==1.4.6
 
-django-storages==1.9.1
+#Azure storage
+django-storages[azure]
```
Comment (krishnamadhavan) [inline requirements.txt:12]: Can we mention a version number to this package to avoid future incompatibilities ?

File: requirements.txt [hunk 6]
```diff
@@ -5,11 +5,12 @@ pytz==2017.2
 # PIL support
 Pillow==7.1.0
 python-dateutil==2.7.5
-
```
Comment (krishnamadhavan) [inline requirements.txt:8]: Can we revert this removal of a new line ?

File: .docker/azure_deployment/Dockerfile.prod.azure (hunk 1 above)
Comment (cjoshi-who) [reply .docker/azure_deployment/Dockerfile.prod.azure:37]: Yes

File: .docker/azure_deployment/settings_local.prod.ecs.py (hunk 2 above)
Comment (cjoshi-who) [reply .docker/azure_deployment/settings_local.prod.ecs.py:150]: Yes

**Review by krishnamadhavan — CHANGES_REQUESTED** [review]
Please make necessary updates as per comments.

File: apps/mainsite/settings.py (hunk 3 above)
Comment (cjoshi-who) [reply apps/mainsite/settings.py:506]: @krishnamadhavan  Only in prod

File: apps/mainsite/settings_local.py.example (hunk 4 above)
Comment (cjoshi-who) [reply apps/mainsite/settings_local.py.example:107]: Shall I remove it from comment or like remove entirely ?

File: requirements.txt (hunk 5 above)
Comment (cjoshi-who) [reply requirements.txt:12]: Okay, will try it up.

File: requirements.txt (hunk 6 above)
Comment (cjoshi-who) [reply requirements.txt:8]: Yes

File: apps/mainsite/settings.py (hunk 3 above)
Comment (krishnamadhavan) [reply apps/mainsite/settings.py:506]: @cjoshi-who In that case, we need it as env variables right ? But here it looks like they are hard-coded with dummy values. Will these variables be overridden somewhere else?

File: apps/mainsite/settings_local.py.example (hunk 4 above)
Comment (krishnamadhavan) [reply apps/mainsite/settings_local.py.example:107]: Since it is commented out and the application is working fine without it, we can remove it entirely.

File: requirements.txt (hunk 5 above)
Comment (krishnamadhavan) [reply requirements.txt:12]: @cjoshi-who We can use the same version as before `1.9.1`

File: apps/mainsite/settings.py (hunk 3 above)
Comment (cjoshi-who) [reply apps/mainsite/settings.py:506]: Yes, settings_local.prod.py will override these variables.

**Review by krishnamadhavan — APPROVED** [review]

File: .docker/azure_deployment/settings_local.prod.azure.py [hunk 7]
```diff
@@ -2,9 +2,8 @@
 
 import random
 import string
-from .settings import *
+from .settings import
```
Comment (krishnamadhavan) [inline .docker/azure_deployment/settings_local.prod.azure.py:5]: Why did you remove the `*` here ?
